### PR TITLE
ci(coverage): re-enable coverage

### PR DESCRIPTION
**Pull request by P403n1x87**
## Description

This PR re-enables coverage support in CI after its disabling in #2414. Coverage is still excluded just for the profile tox scenario since it is still failing.

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

**Comment by P403n1x87** [comment]
Closing for now as there don't seem to be improvements.

File: tox.ini [hunk 1]
```diff
@@ -304,13 +305,14 @@ passenv=
 commands =
 # run only essential tests related to the tracing client
     tracer: pytest {posargs} tests/tracer
-    profile: python -m tests.profiling.run pytest --capture=no --verbose --benchmark-disable {posargs} tests/profiling
+    # profile: python -m tests.profiling.run pytest --capture=no --verbose --benchmark-disable {posargs} tests/profiling
```
Comment (Kyle-Verhoog) [inline tox.ini:308]:
```suggestion
```

I don't really mind leaving it in either but wanted to make sure it was intentional

**Review by Kyle-Verhoog — APPROVED** [review]

File: tox.ini (hunk 1 above)
Comment (P403n1x87) [reply tox.ini:308]: Yes I've intentionally left that there but perhaps would be best to put a comment that explains why we are excluding profile from coverage. WDYT?

Comment (brettlangdon) [reply tox.ini:308]: comment would be nice

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**